backend/src/layer/logic/mealplan_management/meal_plan_manager.rs

```diff
@@ -0,0 +1,103 @@
+use crate::interface::mealplan_management::MensaParseScheduling;
+use crate::interface::mensa_parser::model::ParseCanteen;
+use crate::interface::mensa_parser::MealplanParser;
+use crate::interface::persistent_data::MealplanManagementDataAccess;
+use crate::layer::logic::mealplan_management::relation_resolver::RelationResolver;
+use crate::util::Date;
+use async_trait::async_trait;
+use chrono::Utc;
+use tracing::log::{debug, warn};
+
+pub struct MealPlanManager<Parser, DataAccess>
+where
+    Parser: MealplanParser + Send + Sync,
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+{
+    resolver: RelationResolver<DataAccess>,
+    parser: Parser,
+}
+
+impl<Parser, DataAccess> MealPlanManager<Parser, DataAccess>
+where
+    DataAccess: MealplanManagementDataAccess + Send + Sync,
+    Parser: MealplanParser + Send + Sync,
+{
+    pub const fn _new(database: DataAccess, meal_plan_parser: Parser) -> Self {
+        Self {
+            resolver: RelationResolver::_new(database),
+            parser: meal_plan_parser,
+        }
+    }
+
+    async fn start_resolving(&self, parse_canteens: Vec<ParseCanteen>, date: Date) {
+        for parse_canteen in parse_canteens {
+            let name = &parse_canteen.name.clone();
+            match self.resolver.resolve(parse_canteen, date).await {
+                Ok(_canteen) => debug!("resolved canteen '{name}' with no errors"),
+                Err(error) => warn!("resolved canteen '{name}' with errors: {error}"),
```

Comment:
If a canteen, line or dish can't be resolved this err matching creates a log entry. After that the loop continues the procedure. With this failure management some data can be loss; but the remaining amount 'll be solved. Otherwise the method crashes at this point. Actually you promoted this behaviour. Or I don't understand the mentioned problem?